packages/packagers/css/src/CSSPackager.js

```diff
@@ -99,3 +125,82 @@ export default (new Packager({
     });
   },
 }): Packager);
+
+async function processCSSModule(
+  options,
+  logger,
+  bundleGraph,
+  bundle,
+  asset,
+  media,
+): Promise<[Asset, string, ?Buffer]> {
+  let ast: Root = postcss.fromJSON(nullthrows((await asset.getAST())?.program));
+
+  let usedSymbols = bundleGraph.getUsedSymbols(asset);
+  let localSymbols = new Set(
+    [...asset.symbols].map(([, {local}]) => `.${local}`),
+  );
+
+  let defaultImport = null;
+  if (usedSymbols.has('default')) {
+    let incoming = bundleGraph.getIncomingDependencies(asset);
+    // `import * as ns from ""; ns.default` is fine.
+    defaultImport = incoming.find(d => d.meta.hasDefaultImport);
+    if (defaultImport) {
+      let loc = defaultImport.symbols.get('default')?.loc;
+      logger.warn({
+        message:
+          'CSS modules cannot be tree shaken when imported with a default specifier',
```

Comment:
We could get rid of this limitation if we tracked member accesses of default/named imports like we were just discussing 😉